part 1

Nutsheil opened the pull request.

Commit:
added primitives

Commit:
added marked dataset

Nutsheil committed:
Update README.md

Nutsheil committed:
Create README.md

Nutsheil committed:
Update README.md

Nutsheil committed:
Update README.md

Nutsheil committed:
Update README.md

Nutsheil committed:
Delete 8.jpg

Nutsheil committed:
Delete 9.jpg

Nutsheil committed:
Delete 10.jpg

Nutsheil committed:
Delete 7.jpg

Nutsheil committed:
Add files via upload

Nutsheil committed:
Delete 7.jpg

Nutsheil committed:
Add files via upload

Nutsheil committed:
Delete 7.jpg

Nutsheil committed:
Update README.md

Nutsheil committed:
Add files via upload

Nutsheil committed:
Update README.md

Nutsheil committed:
Update README.md

Nutsheil committed:
Delete 6.jpg

Nutsheil committed:
Add files via upload

Nutsheil committed:
Update README.md

Nutsheil committed:
problem solving plan

Nutsheil committed:
Update README.md

BaevDaniil reviewed (commented):
* Хотелось бы больше деталей в плане. Например, более детальный алгоритм поиска многоугольника и объектов, дискретность проверяемых положений, возможно критерий качества расположения объектов для сокращения перебора
* Наверное, лучше перенести весь датасет в соответствующую папку и уже внутри разбить на подпапки test, true, false
* Добавить в ридми ссылки на указанные ресурсы

BaevDaniil commented on README.md:
```diff
@@ -1 +1,61 @@
+* Фотографии не должны быть никак обработаны перед использованием
+* Объекты и лист А4 с нарисованным многоугольником должны быть размещены на поверхности из тренировочного набора данных
+
+### Требования к многоугольнику
```
Есть ли ограничения на кол-во вершин многоугольника?

Nutsheil committed:
create new folder

Nutsheil committed:
Add files via upload

Nutsheil committed:
Delete 9.jpg

Nutsheil committed:
Delete 0.jpg

Nutsheil committed:
Delete 1.jpg

Nutsheil committed:
Delete 10.jpg

Nutsheil committed:
Delete 11.jpg

Nutsheil committed:
Delete 2.jpg

Nutsheil committed:
Delete 3.jpg

Nutsheil committed:
Delete 4.jpg

Nutsheil committed:
Delete 5.jpg

Nutsheil committed:
Delete 6.jpg

Nutsheil committed:
Delete 7.jpg

Nutsheil committed:
Delete 8.jpg

Nutsheil committed:
Update README.md

Nutsheil committed:
Add files via upload

Nutsheil committed:
Delete README.md

Nutsheil committed:
Delete dataset/false directory

Nutsheil committed:
Create README.md

Nutsheil committed:
Add files via upload

Nutsheil committed:
Delete dataset/true directory

Nutsheil committed:
Create README.md

Nutsheil committed:
Add files via upload

Nutsheil committed:
Delete primitives directory

Nutsheil committed:
Create README.md

Nutsheil committed:
Add files via upload

Nutsheil committed:
Update README.md

Nutsheil committed:
Update README.md

Nutsheil committed:
Update README.md

ValyMihalchuk commented on README.md:
```diff
@@ -1 +1,63 @@
\ No newline at end of file
+# intellegent_placer
+## Постановка задачи
+Необходимо определить по фотографии предмета(-ов) и фотографии многоугольника, который нарисован черным маркером на белом листе A4, могут ли данные предметы поместиться в этот многоугольник. Все предметы, которые могут использоваться, заранее известны.
```
Насколько я понимаю, на самом деле речь идет об одной фотографии) то есть фотография одна, а на ней предметы + многоугольник

ValyMihalchuk commented on README.md:
```diff
@@ -1 +1,63 @@
+
+### Ввод и вывод
+* Программа получает на вход путь до фотографии со всеми объектами и нарисованным многоугольником
+* В случае если входные данные корректны, программа может ввести два варианта ответов: либо True (предметы можно поместить в многоугольник) либо False (предметы нельзя поместить в многоугольник). В случае некорректных данных программа выводит False. Корректность данных проверяется исходя из требований, которые приведены ниже
```
В случае некорректных данных программа будет выдавать False, корректность в свою очередь, как сказано, определяется требованиями, но на самом деле, если посмотреть на требования ниже, то проверять корректность их программой довольно сложно. Или если действительно проверять, то стоит описать это в плане. Например требование про необработанность фотографии- мне кажется это невозможно проверить) или про перпендикулярность камеры. Короче, думаю, стоит ослабить требование выдавать False на _всех_ некорректных данных.

ValyMihalchuk commented on README.md:
```diff
@@ -1 +1,63 @@
+* Дыры, находящиеся внутри объектов, не учитываются в работе программы, в них нельзя будет поместить объекты
+
+# План решения задачи
+1. Найти на изображении белый лист бумаги формата А4
```
вот тут на самом деле, хотелось бы подробнее, как будет искаться лист)
Во-первых, если посмотреть на примеры (true/false), их там два, и они перекрываются - тот, на котором лежат предметы, и там где многоугольник. Или считаем это все за один лист? Но тогда это не А4)
И очень важный момент - помимо этих листов там еще поверхность, на которой они все лежат. Хотелось бы тут увидеть метод, как лист/листы будет от нее отделяться

ValyMihalchuk replied:
И сюда же: в папке primitives есть изображение surface.jpg - то есть изображение поверхности... Но на самом деле, это не поверхность, а два белых листа(
Если это под этим действительно подразумевается вся поверхность, то чтобы в этом был смысл, лучше тогда обрезать фотографии только до нее, как мне кажется

ValyMihalchuk commented on README.md:
```diff
@@ -1 +1,63 @@
+
+# План решения задачи
+1. Найти на изображении белый лист бумаги формата А4
+2. Найти на белом листе бумаги многоугольник, нарисованный черными маркером. Предполагается искать его по цвету маркера
```
Тут все ок, просто хочу еще обратить внимание на вот что: как я понял исходя из тестового набора, там и предметы и многоугольник на белых листах а4, причем они перекрываются - вот это немножко напрягает, не будет ли это перекрытие считываться как граница, и если будет, то как с этим бороться.

ValyMihalchuk commented on README.md:
```diff
@@ -1 +1,63 @@
+# План решения задачи
+1. Найти на изображении белый лист бумаги формата А4
+2. Найти на белом листе бумаги многоугольник, нарисованный черными маркером. Предполагается искать его по цвету маркера
+3. Найти на изображении все остальные объекты, находящиеся ниже белого листа бумаги. Предполагается искать их по сильному контрасту с фоном или наличию "шумов" на краях объектов
```
Стоп))) тут не понял - почему ниже белого листа? Они же все вверху и на белом листе (это если исходить из примеров)???

а если из требованиий:
> Объекты должны распологаться выше нарисованного многоугольника (выше верхней точки многоугольника)

то есть и тут выше)

ValyMihalchuk reviewed (commented):
По поводу требований добавил пару мелких замечаний, в основном там все более менее ок. 
Еще возникло пару вопросов к примерам, и то, как они соответствуют плану и требованиям

Nutsheil committed:
Update README.md

Nutsheil committed:
Add files via upload

Nutsheil committed:
Delete 0.jpg

Nutsheil committed:
Delete 1.jpg

Nutsheil committed:
Delete 10.jpg

Nutsheil committed:
Delete 11.jpg

Nutsheil committed:
Delete 2.jpg

Nutsheil committed:
Delete 3.jpg

Nutsheil committed:
Delete 4.jpg

Nutsheil committed:
Delete 5.jpg

Nutsheil committed:
Delete 6.jpg

Nutsheil committed:
Delete 7.jpg

Nutsheil committed:
Delete 8.jpg

Nutsheil committed:
Delete 9.jpg

Nutsheil committed:
Add files via upload

Nutsheil committed:
Delete 0.jpg

Nutsheil committed:
Add files via upload

Nutsheil committed:
Iteration 1 completed

work has been done on the most representative tests
in the course of this work, masks were obtained and the boundaries of polygons with objects were found
work continues!

Nutsheil committed:
Add files via upload

Nutsheil committed:
Update README.md

Nutsheil committed:
first auto commit from fucking PyCharm

Nutsheil committed:
move files

Nutsheil committed:
Update README.md

Nutsheil committed:
first step to replace project architecture

Nutsheil committed:
add new files

Nutsheil committed:
Structure of components. Split picture on items and polygon. Read primitives. First algorithm

Nutsheil committed:
second iteration (demo ver)

Nutsheil committed:
update logic of show items and entity "picture"

Nutsheil committed:
Update README.md

Nutsheil committed:
little fixes, it doesn't matter

ValyMihalchuk commented on intelligent_placer_lib/contours.py:
```diff
@@ -26,14 +28,17 @@ def get_contours_v1(image):
+            for x in range(x_most_left, x_most_left + width):
+                if cv2.pointPolygonTest(cnt, (y, x), False) >= 0:
+                    mask[x - x_most_left][y - y_most_bottom] = True
+        masks.append((mask * 255).astype("uint8"))
```
255 лучше как константу вынести :)

Nutsheil replied:
Да, согласен
Уже исправил
Обновлю в следующем коммите в ближайшее время

ValyMihalchuk commented on intelligent_placer_lib/intelligent_placer.py:
```diff
@@ -24,31 +24,26 @@ def split_items_and_polygon(picture, contours):
-        image = compress_image(image, 0.2)
-        contour = get_contours_v1(image)
+    for picture in pictures:
+        picture.image = compress_image(picture.image, 0.2)
```
0.2 - аналогично, в константу

ValyMihalchuk reviewed (commented):
этап работы с нахождением многоугольника и предметов выполнен, код написан хорошо, небольшие пожелания добавил
